========= 512 
==========df 

&& 2.484076738357544

N_seg 2

&&  2.494990587234497

N_seg 3

&&  2.5219979286193848

N_seg 4

&&  2.5000312328338623

N_seg 5

&&  2.497981309890747

N_seg 6

&&  2.4851064682006836

N_seg 7

&&  2.45512318611145

N_seg 8

&&  2.5418219566345215

N_seg 9

&&  2.4679930210113525

N_seg 10

&&  2.4793972969055176

========= 1024 

&& 2.5426323413848877

N_seg 2

&&  2.5567212104797363

N_seg 3

&&  2.5384092330932617

N_seg 4

&&  2.587198257446289

N_seg 5

&&  2.5841832160949707

N_seg 6

&&  2.5511627197265625

N_seg 7

&&  2.5578529834747314

N_seg 8

&&  2.5598061084747314

N_seg 9

&&  2.5391972064971924

N_seg 10

&&  2.5549981594085693

========= 2048 

&& 2.7799057960510254

N_seg 2

&&  2.820913553237915

N_seg 3

&&  2.8191142082214355

N_seg 4

&&  2.795236110687256

N_seg 5

&&  2.788604974746704

N_seg 6

&&  2.8106021881103516

N_seg 7

&&  2.789832353591919

N_seg 8

&&  2.8203628063201904

N_seg 9

&&  2.800868034362793

N_seg 10

&&  2.833585739135742

========= 3072 

&& 3.1888997554779053

N_seg 2

&&  3.087562322616577

N_seg 3

&&  3.082118272781372

N_seg 4

&&  3.124223232269287

N_seg 5

&&  3.112614393234253

N_seg 6

&&  3.0704572200775146

N_seg 7

&&  3.1018354892730713

N_seg 8

&&  3.0843284130096436

N_seg 9

&&  3.1226515769958496

N_seg 10

&&  3.126405715942383

========= 4096 
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 174, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 132, in forward
    out = self.block1(x)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/container.py", line 119, in forward
    input = module(input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 512.00 MiB (GPU 0; 39.59 GiB total capacity; 37.50 GiB already allocated; 352.19 MiB free; 37.76 GiB reserved in total by PyTorch)
N_seg 2

&&  4.033070802688599

N_seg 3

&&  3.848970651626587

N_seg 4

&&  3.591827392578125

N_seg 5

&&  3.580253839492798

N_seg 6

&&  3.5892579555511475

N_seg 7

&&  3.725555896759033

N_seg 8

&&  3.6117630004882812

N_seg 9

&&  3.546363592147827

N_seg 10

&&  3.5951035022735596

========= 5120 
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 174, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 132, in forward
    out = self.block1(x)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/container.py", line 119, in forward
    input = module(input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 3.12 GiB (GPU 0; 39.59 GiB total capacity; 36.29 GiB already allocated; 1.81 GiB free; 36.29 GiB reserved in total by PyTorch)
N_seg 2
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 181, in main
    loss.backward()
  File "/home/yufan/labpytorch/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
  File "/home/yufan/labpytorch/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 98, in backward
    outputs = ctx.run_function(*detached_inputs)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 1.56 GiB (GPU 0; 39.59 GiB total capacity; 34.39 GiB already allocated; 1.09 GiB free; 36.44 GiB reserved in total by PyTorch)
N_seg 3
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 181, in main
    loss.backward()
  File "/home/yufan/labpytorch/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
  File "/home/yufan/labpytorch/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 114, in backward
    torch.autograd.backward(outputs_with_grad, args_with_grad)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
RuntimeError: CUDA out of memory. Tried to allocate 3.12 GiB (GPU 0; 39.59 GiB total capacity; 33.22 GiB already allocated; 2.65 GiB free; 34.88 GiB reserved in total by PyTorch)
N_seg 4
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 181, in main
    loss.backward()
  File "/home/yufan/labpytorch/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
  File "/home/yufan/labpytorch/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 114, in backward
    torch.autograd.backward(outputs_with_grad, args_with_grad)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
RuntimeError: CUDA out of memory. Tried to allocate 6.25 GiB (GPU 0; 39.59 GiB total capacity; 30.10 GiB already allocated; 2.63 GiB free; 34.90 GiB reserved in total by PyTorch)
N_seg 5
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 181, in main
    loss.backward()
  File "/home/yufan/labpytorch/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
  File "/home/yufan/labpytorch/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 114, in backward
    torch.autograd.backward(outputs_with_grad, args_with_grad)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
RuntimeError: CUDA out of memory. Tried to allocate 6.25 GiB (GPU 0; 39.59 GiB total capacity; 25.41 GiB already allocated; 5.76 GiB free; 31.77 GiB reserved in total by PyTorch)
N_seg 6
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 181, in main
    loss.backward()
  File "/home/yufan/labpytorch/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
  File "/home/yufan/labpytorch/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 114, in backward
    torch.autograd.backward(outputs_with_grad, args_with_grad)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
RuntimeError: CUDA out of memory. Tried to allocate 6.25 GiB (GPU 0; 39.59 GiB total capacity; 25.41 GiB already allocated; 2.64 GiB free; 34.90 GiB reserved in total by PyTorch)
N_seg 7
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 181, in main
    loss.backward()
  File "/home/yufan/labpytorch/torch/tensor.py", line 245, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
  File "/home/yufan/labpytorch/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 114, in backward
    torch.autograd.backward(outputs_with_grad, args_with_grad)
  File "/home/yufan/labpytorch/torch/autograd/__init__.py", line 146, in backward
    Variable._execution_engine.run_backward(
RuntimeError: CUDA out of memory. Tried to allocate 6.25 GiB (GPU 0; 39.59 GiB total capacity; 25.41 GiB already allocated; 2.64 GiB free; 34.90 GiB reserved in total by PyTorch)
N_seg 8

&&  6.714999675750732

N_seg 9

&&  6.9689881801605225

N_seg 10

&&  6.9509148597717285

========= 6144 
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 174, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 132, in forward
    out = self.block1(x)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/container.py", line 119, in forward
    input = module(input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 31.28 GiB already allocated; 7.04 GiB free; 31.28 GiB reserved in total by PyTorch)
N_seg 2
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 3
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 4
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 5
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 6
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 7
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 8
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 9
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
N_seg 10
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 81.00 GiB (GPU 0; 39.59 GiB total capacity; 18.48 GiB already allocated; 19.84 GiB free; 18.48 GiB reserved in total by PyTorch)
========= 7168 
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 174, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-no-cp.py", line 132, in forward
    out = self.block1(x)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/container.py", line 119, in forward
    input = module(input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 12.25 GiB (GPU 0; 39.59 GiB total capacity; 30.30 GiB already allocated; 8.02 GiB free; 30.30 GiB reserved in total by PyTorch)
N_seg 2
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 3
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 4
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 5
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 6
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 7
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 8
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 9
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
N_seg 10
Traceback (most recent call last):
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 223, in <module>
    main()
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 168, in main
    out_ref = model_ref(input_ref)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/uu/benchmarking/vgg16-ref-sq-cp.py", line 132, in forward
    out = checkpoint_sequential(self.block1, N_seg, x)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 243, in checkpoint_sequential
    input = checkpoint(run_function(start, end, functions), input,
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 179, in checkpoint
    return CheckpointFunction.apply(function, preserve, *args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 75, in forward
    outputs = run_function(*args)
  File "/home/yufan/labpytorch/torch/utils/checkpoint.py", line 230, in forward
    input = functions[j](input)
  File "/home/yufan/labpytorch/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 399, in forward
    return self._conv_forward(input, self.weight, self.bias)
  File "/home/yufan/labpytorch/torch/nn/modules/conv.py", line 395, in _conv_forward
    return F.conv2d(input, weight, bias, self.stride,
RuntimeError: CUDA out of memory. Tried to allocate 110.25 GiB (GPU 0; 39.59 GiB total capacity; 25.13 GiB already allocated; 13.19 GiB free; 25.13 GiB reserved in total by PyTorch)
